.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "_deps/gtest-src"]
```

Comment:
How can I resolve this?

Reply:
Just delete all your additions. git rm `_deps/gtest-src`

Reply:
I've been grappling with another issue for the past few days, but despite my efforts, I haven't been able to identify its root cause.

Reply:
What is the issue?

Reply:
Except 2 all checks are failing just tell me a approach I will try to fix it

Reply:
You will understand how to fix it when you build this tool locally. This can be a good learning experience. 

The fix is actually really simple, you tried to merge and accidentally left some merge text in the CPP files.

Reply:
I successfully built the code locally and then ran make -j4 check, confirming that all test cases passed.

Reply:
I think you may have the wrong build, this can not work with  broken changes. What are the exact steps you are using?

Reply:
I have followed this commands : 
mkdir build
cd build
cmake .. 
make -j4
make -j4 check  
result : 
100% tests passed, 0 tests failed out of 2752

Label Time Summary:
XFAIL                              =  57.02 sec*proc (34 tests)
bmv2                               = 2012.04 sec*proc (567 tests)
bmv2-parser-inline-opt-disabled    =  45.01 sec*proc (13 tests)
bmv2-parser-inline-opt-enabled     =  43.97 sec*proc (13 tests)
bmv2-ptf                           =  20.21 sec*proc (1 test)
dpdk                               = 311.44 sec*proc (249 tests)
ebpf                               =  74.33 sec*proc (28 tests)
ebpf-bcc                           =  18.16 sec*proc (26 tests)
ebpf-errors                        =   0.67 sec*proc (1 test)
ebpf-kernel                        =  38.44 sec*proc (30 tests)
err                                =  86.05 sec*proc (319 tests)
graph                              =   2.03 sec*proc (2 tests)
gtest                              =  38.01 sec*proc (1 test)
p14_to_16                          = 520.84 sec*proc (194 tests)
p4                                 = 1780.73 sec*proc (1221 tests)
p4tc                               =  36.47 sec*proc (51 tests)
p4unroll                           =  25.34 sec*proc (13 tests)
ubpf                               =  36.62 sec*proc (19 tests)

Total Test time (real) = 425.09 sec
make[3]: Leaving directory '/home/sumit/p4c-1/build'
Built target check-all
make[3]: Entering directory '/home/sumit/p4c-1/build'
Scanning dependencies of target check
make[3]: Leaving directory '/home/sumit/p4c-1/build'
Built target check
make[2]: Leaving directory '/home/sumit/p4c-1/build'
make[1]: Leaving directory '/home/sumit/p4c-1/build'

Reply:
What does git status say.

Reply:
git status
On branch comment-formatting-involved-dpdk
Your branch is ahead of 'origin/comment-formatting-involved-dpdk' by 1 commit.
  (use "git push" to publish your local commits)

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        _deps/
        bmv2-parser-inline-opt-disabled/
        bmv2-parser-inline-opt-enabled/
        bmv2-ptf/
        bmv2/
        control-plane/CMakeFiles/
        control-plane/CTestTestfile.cmake
        control-plane/Makefile
        control-plane/cmake_install.cmake
        dpdk/
        ebpf-bcc/
        ebpf-errors/
        ebpf-kernel/
        ebpf/
        err/
        frontends/CMakeFiles/
        frontends/CTestTestfile.cmake
        frontends/Makefile
        frontends/cmake_install.cmake
        graph/
        ir/CMakeFiles/
        ir/CTestTestfile.cmake
        ir/Makefile
        ir/cmake_install.cmake
        lib/CMakeFiles/
        lib/CTestTestfile.cmake
        lib/Makefile
        lib/cmake_install.cmake
        midend/CMakeFiles/
        midend/CTestTestfile.cmake
        midend/Makefile
        midend/cmake_install.cmake
        p14_to_16/
        p4/
        p4tc/
        p4unroll/
        test/CMakeFiles/
        test/CTestTestfile.cmake
        test/Makefile
        test/cmake_install.cmake
        test/gtest/env.h
        tools/driver/CMakeFiles/
        tools/driver/CTestTestfile.cmake
        tools/driver/Makefile
        tools/driver/cmake_install.cmake
        tools/ir-generator/CMakeFiles/
        tools/ir-generator/CTestTestfile.cmake
        tools/ir-generator/Makefile
        tools/ir-generator/cmake_install.cmake
        ubpf/

nothing added to commit but untracked files present (use "git add" to track)

git log :
commit 35e298c298b21333d800bfa1f5f891312b6a93cc (HEAD -> comment-formatting-involved-dpdk)
Author: sumit7754 <sumitjain7754@gmail.com>
Date:   Thu Apr 4 21:15:08 2024 +0530

    fix the typo

commit 33a6ace3f39b074ba5f1a01f4f2f8b10381ef2e7 (origin/comment-formatting-involved-dpdk)
Author: sumit7754 <sumitjain7754@gmail.com>
Date:   Thu Apr 4 20:06:57 2024 +0530

    Remove submodule _deps/gtest-src

commit 3b7caa47fa99146a42d46ffc2a3a06cdb04cd666
Author: sumit7754 <sumitjain7754@gmail.com>
Date:   Thu Apr 4 19:37:28 2024 +0530

    .

commit 3495945b4f1b49ce8bc93f2d877477f45914a92a
Author: sumit7754 <sumitjain7754@gmail.com>
Date:   Thu Apr 4 19:37:17 2024 +0530

Reply:
There is something really broken, why are all these files untracked?

Reply:
I have a guess, but only a guess.  I believe I have seen behavior like this when I have a Linux VM running on a host OS like macOS or Windows, and I do a clone in a directory within the VM that is shared with a folder on the host OS.

If that is the case, try running `git status` in the host OS directory and see if it is any different.

Reply:
I'm currently working within WSL, and I've cloned the directory in the Ubuntu environment. Can you provide further guidance?

Reply:
I have never seen behavior like this when I create a Linux VM inside of VirtualBox on Windows, or inside of VirtualBox or UTM on macOS, and keep all files local inside of the VM's file system.  I have no personal experience with WSL to share, not having used it.

Reply:
I'm looking for further suggestions to resolve this issue. Any guidance or recommendations would be greatly appreciated

Reply:
One possibility is to abandon one PR, and start fresh from a new clone and create a new PR.  I do not recommend this as the first option, but it is one way to make progress if you cannot find any other way.

Reply:
Another is to consider using VirtualBox to create a Linux VM, and do all of your work of creating PRs inside of that.

Reply:
Thanks @jafingerhut for the helpful suggestions. It turns out that WSL was causing issues for me. By switching to building the project in VirtualBox, I've managed to resolve the problem.

Reply:
@fruffy Can I close this pr and do the work from start?

Reply:
Sure, if that is easier for you.